### PR TITLE
Complete refactoring of Cave

### PR DESCRIPTION
**Pull request by jbonofre**

### CODE REVIEW

**Comment by jbonofre** [comment]
retest this please

**Comment by jbonofre** [comment]
retest this please

**Comment by jbonofre** [comment]
retest this please

File: deployer/service/src/main/java/org/apache/karaf/cave/deployer/service/rest/DeployerRestService.java [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    private DeployerService deployerService;
+
+    @Reference
+    private HttpService httpService;
+
+    @Activate
+    public void activate(ComponentContext componentContext) throws Exception {
+        DeployerRestApi restApi = new DeployerRestApi(deployerService);
+        DeployerRestServlet featuresGatewayRestServlet = new DeployerRestServlet(restApi);
+        httpService.registerServlet("/cave/deployer/api", featuresGatewayRestServlet, null, null);
```
Comment (fpapon) [inline deployer/service/src/main/java/org/apache/karaf/cave/deployer/service/rest/DeployerRestService.java]: We should name the variable `deployerRestServlet`.

Comment (jbonofre) [reply deployer/service/src/main/java/org/apache/karaf/cave/deployer/service/rest/DeployerRestService.java]: +1, copy/paste mistake. I'm fixing.

**Comment by jbonofre** [comment]
@fpapon I'm fixing javadoc and an artifact upload warning. Gonna push shortly.

File: pom.xml [hunk 2]
```diff
@@ -30,18 +30,19 @@
...
 
     <properties>
         <commons-io.version>2.6</commons-io.version>
-        <cxf.version>3.1.18</cxf.version>
-        <httpclient.version>4.3.4</httpclient.version>
-        <jsoup.version>1.7.3</jsoup.version>
-        <karaf.version>4.1.7</karaf.version>
+        <cxf.version>3.3.2</cxf.version>
+        <httpclient.version>4.5.10</httpclient.version>
+        <jackson.version>2.9.10</jackson.version>
```
Comment (coheigea) [inline pom.xml]: Could you update this to 2.10.0?

Comment (jbonofre) [reply pom.xml]: Sure !

**Comment by jbonofre** [comment]
I'm fixing itest on Jenkins (need to use the Jenkins local repository).

**Comment by jbonofre** [comment]
I also detected a conflict between the different REST endpoints, I'm fixing that.

**Comment by jbonofre** [comment]
I changed Jenkins configuration about the local Maven repository.

**Comment by jbonofre** [comment]
retest this please

**Comment by jbonofre** [comment]
retest this please

**Comment by jbonofre** [comment]
Finalized including repository scheduler.

**Pull request merged**